File: frontend/src/models/activity.ts
import { autoserializeAs, deserializeAs } from "dcerialize";

/**
 * Objective model
 */
export class Activity {
    /**
     * Activity name
     */
    @autoserializeAs(() => String) name: string;

    /**
     * Activity description
     */
    @autoserializeAs(() => String) description: string | undefined;

    /**
     * Activity ID
     */
    @deserializeAs(() => Number) id: number | undefined;

    /**
     * Category ID
     */
    @autoserializeAs(() => Number, 'category') category: number | undefined;

    /**
     * Constructor
     */
    constructor(
        name: string,
        description?: string,
        category?: number,
        id?: number,
    ) {
        this.name = name;
        this.description = description;
        this.id = id;
        this.category = category;
    }
}

File: backend/core/models/activity.py
from ..utils import get_next_reset_day, get_week
from django.db import models
from datetime import timedelta

class Activity(models.Model):
    name = models.CharField(max_length=120)
    description = models.CharField(max_length=500, null=True)
    category = models.ForeignKey(
        'Category',
        default=None,
        null=True,
        on_delete=models.SET_NULL
    )

    def __str__(self):
        return self.name